--- a/drivers/clk/rockchip/clk_rk3399.c
+++ b/drivers/clk/rockchip/clk_rk3399.c
@@ -55,7 +55,7 @@
 
 static const struct pll_div gpll_init_cfg = PLL_DIVISORS(GPLL_HZ, 2, 2, 1);
 static const struct pll_div cpll_init_cfg = PLL_DIVISORS(CPLL_HZ, 1, 2, 2);
-#if !defined(CONFIG_XPL_BUILD)
+#if !defined(CONFIG_SPL_BUILD)
 static const struct pll_div ppll_init_cfg = PLL_DIVISORS(PPLL_HZ, 2, 2, 1);
 #endif
 
@@ -982,6 +982,8 @@
 		break;
 	case DCLK_VOP0:
 	case DCLK_VOP1:
+	case ACLK_VOP0:
+	case HCLK_VOP0:
 		break;
 	case PCLK_EFUSE1024NS:
 		break;
@@ -997,6 +999,8 @@
 	case PCLK_DDR:
 	case ACLK_VDU:
 		break;
+	case SCLK_DPHY_PLL:
+		return 24000000;
 	case PCLK_ALIVE:
 	case PCLK_WDT:
 		rate = rk3399_alive_get_clk(priv->cru);
@@ -1064,6 +1068,8 @@
 	case DCLK_VOP1:
 		ret = rk3399_vop_set_clk(priv->cru, clk->id, rate);
 		break;
+	case ACLK_VOP0:
+	case HCLK_VOP0:
 	case ACLK_VOP1:
 	case HCLK_VOP1:
 	case HCLK_SD:
@@ -1168,8 +1174,14 @@
 		return rk3399_gmac_set_parent(clk, parent);
 	case SCLK_PCIEPHY_REF:
 		return rk3399_pciephy_set_parent(clk, parent);
+	case DCLK_VOP1_DIV:
+	case DCLK_VOP0_DIV:
+	case DCLK_VOP1:
+	case DCLK_VOP0:
+		return 0;
 	}
 
+
 	debug("%s: unsupported clk %ld\n", __func__, clk->id);
 	return -ENOENT;
 }
@@ -1261,6 +1273,12 @@
 		if (readl(&priv->cru->clksel_con[18]) & BIT(10))
 			rk_clrreg(&priv->cru->clkgate_con[12], BIT(6));
 		break;
+	case SCLK_DPHY_TX0_CFG:
+		rk_clrreg(&priv->cru->clkgate_con[21], BIT(1));
+		break;
+	case PCLK_VIO_GRF:
+		rk_clrreg(&priv->cru->clkgate_con[29], BIT(12));
+		break;
 	default:
 		debug("%s: unsupported clk %ld\n", __func__, clk->id);
 		return -ENOENT;
@@ -1356,6 +1374,12 @@
 		if (readl(&priv->cru->clksel_con[18]) & BIT(10))
 			rk_setreg(&priv->cru->clkgate_con[12], BIT(6));
 		break;
+	case SCLK_DPHY_TX0_CFG:
+		rk_setreg(&priv->cru->clkgate_con[21], BIT(1));
+		break;
+	case PCLK_VIO_GRF:
+		rk_setreg(&priv->cru->clkgate_con[29], BIT(12));
+		break;
 	default:
 		debug("%s: unsupported clk %ld\n", __func__, clk->id);
 		return -ENOENT;
@@ -1450,6 +1474,27 @@
 		     pclk_div << PCLK_PERILP1_DIV_CON_SHIFT |
 		     hclk_div << HCLK_PERILP1_DIV_CON_SHIFT |
 		     HCLK_PERILP1_PLL_SEL_GPLL << HCLK_PERILP1_PLL_SEL_SHIFT);
+		     
+		     
+      	/*
+	 * VOP clocks
+	 *
+	 * cpll (800 MHz) -> aclk (400 MHz) -> hclk (100 MHz)
+	 * gpll (594 MHz) -> dclk (74.25 MHz)
+	 *
+	 * DCLK_VOP1 ?
+	 *
+	 * CRU_CLKSEL_CON47  vop0
+	 * CRU_CLKSEL_CON48  vop1
+	 */
+	rk_clrsetreg(&cru->clksel_con[47], 0xffff, 0x340); // aclk/1  hclk/4
+	rk_clrsetreg(&cru->clksel_con[48], 0xffff, 0x340); // aclk/1  hclk/4
+	rk_clrsetreg(&cru->clksel_con[49], 0xffff, 0x207); // gpll/8
+	rk_clrsetreg(&cru->clksel_con[50], 0xffff, 0x207); // gpll/8
+
+	// vop1 (BIT(13) | BIT(11) | BIT(10))
+	// vop0 (BIT(12) | BIT(9) | BIT(8))
+	//rk_setreg(&cru->clkgate_con[10], BIT(12) | BIT(9) | BIT(8));
 }
 
 static int rk3399_clk_probe(struct udevice *dev)
@@ -1463,7 +1508,7 @@
 	priv->cru = map_sysmem(plat->dtd.reg[0], plat->dtd.reg[1]);
 #endif
 
-#if defined(CONFIG_XPL_BUILD)
+#if defined(CONFIG_SPL_BUILD)
 	init_clocks = true;
 #elif CONFIG_IS_ENABLED(HANDOFF)
 	if (!(gd->flags & GD_FLG_RELOC)) {
@@ -1657,7 +1702,7 @@
 	.set_rate = rk3399_pmuclk_set_rate,
 };
 
-#ifndef CONFIG_XPL_BUILD
+#ifndef CONFIG_SPL_BUILD
 static void pmuclk_init(struct rk3399_pmucru *pmucru)
 {
 	u32 pclk_div;
@@ -1675,7 +1720,7 @@
 
 static int rk3399_pmuclk_probe(struct udevice *dev)
 {
-#if CONFIG_IS_ENABLED(OF_PLATDATA) || !defined(CONFIG_XPL_BUILD)
+#if CONFIG_IS_ENABLED(OF_PLATDATA) || !defined(CONFIG_SPL_BUILD)
 	struct rk3399_pmuclk_priv *priv = dev_get_priv(dev);
 #endif
 
@@ -1685,7 +1730,7 @@
 	priv->pmucru = map_sysmem(plat->dtd.reg[0], plat->dtd.reg[1]);
 #endif
 
-#ifndef CONFIG_XPL_BUILD
+#ifndef CONFIG_SPL_BUILD
 	pmuclk_init(priv->pmucru);
 #endif
 	return 0;
