diff --git a/drivers/video/video-uclass.c b/drivers/video/video-uclass.c
index ff4f2199..da1aa7c3 100644
--- a/drivers/video/video-uclass.c
+++ b/drivers/video/video-uclass.c
@@ -532,7 +532,7 @@ static int show_splash(struct udevice *dev)
 	u8 *data = SPLASH_START(u_boot_logo);
 	int ret;
 
-	ret = video_bmp_display(dev, map_to_sysmem(data), -4, 4, true);
+	ret = video_bmp_display(dev, map_to_sysmem(data), -4, 135, true);
 
 	return 0;
 }
